packages/firestore/test/unit/specs/existence_filter_spec.test.ts

```diff
@@ -286,4 +246,347 @@ describeSpec('Existence Filters:', [], () => {
+          .expectLimboDocs()
+      );
+    }
+  );
+
+  specTest(
+    'Bloom filter can process special characters in document name',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/ÀÒ∑', 1000, { v: 1 });
+      const docB = doc('collection/À∑Ò', 1000, { v: 1 });
+      const bloomFilterProto = generateBloomFilterProto({
+        contains: [docA],
+        notContains: [docB],
+        numOfBits: 2,
+        hashCount: 1
+      });
+      return (
+        spec()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA, docB)
+          .expectEvents(query1, { added: [docA, docB] })
+          // DocB is deleted in the next sync.
+          .watchFilters([query1], [docA.key], bloomFilterProto)
+          .watchSnapshots(2000)
+          // BloomFilter identifies docB is deleted, skip full query and put
+          // docB into limbo directly.
+          .expectEvents(query1, { fromCache: true })
+          .expectLimboDocs(docB.key) // DocB is now in limbo.
+      );
+    }
+  );
+
+  specTest(
+    'Bloom filter fills in default values for undefined padding and hashCount',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/a', 1000, { v: 1 });
+      const docB = doc('collection/b', 1000, { v: 2 });
+
+      return (
+        spec()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA, docB)
+          .expectEvents(query1, { added: [docA, docB] })
+          // DocB is deleted in the next sync.
+          .watchFilters([query1], [docA.key], {
+            // Bloom filter omitted padding and hashCount. Default value 0 will
+            // be used.
+            bits: { bitmap: 'AQ==' }
+          })
+          .watchSnapshots(2000)
+          // Re-run query is triggered.
+          .expectEvents(query1, { fromCache: true })
+          .expectActiveTargets({ query: query1, resumeToken: '' })
+      );
+    }
+  );
+
+  specTest(
+    'Full re-query is triggered when bloom filter bitmap is invalid',
+    [],
+    () => {
+      const query1 = query('collection');
+      const docA = doc('collection/a', 1000, { v: 1 });
+      const docB = doc('collection/b', 1000, { v: 1 });
+
+      return (
+        spec()
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, docA, docB)
+          .expectEvents(query1, { added: [docA, docB] })
+          // DocB is deleted in the next sync.
+          .watchFilters([query1], [docA.key], {
+            // Invalid base64 string in bitmap.
+            bits: { bitmap: 'INVALID_BASE_64', padding: 1 },
+            hashCount: 1
+          })
+          .watchSnapshots(2000)
+          // Re-run query is triggered.
+          .expectEvents(query1, { fromCache: true })
+          .expectActiveTargets({ query: query1, resumeToken: '' })
+      );
+    }
+  );
+
+  specTest(
+    'Full re-query is triggered when bloom filter input is invalid',
```

Comment:
Update the test name "bloom filter input is invalid" to specify _how_ it is invalid. In this case, the test is testing a negative hashCount.